### DISCUSSION TITLE
"[json.exception.type_error.302] type must be number, but is null"}

### DISCUSSION THREAD

**Discussion by liangzhang01** [nlohmann/json#3852] [Q&A]
Hello,

i got an issue about "[json.exception.type_error.302] type must be number, but is null"}, need your help to advise if anythong wrong, much appreciated!
My json string is like:

```json
{
  "NoBL": 2,
  "NoAL": 2,
 "BL":[
   {"Price":125, "Qty":8,"Num":6},
   {"Price":126, "Qty":8,"Num":6}
 ],
 "AL":[
   {"Price":127, "Qty":8,"Num":6},
   {"Price":128, "Qty":8,"Num":6}
 ]
}
```

and the code snippet is: 

```cpp
         auto data= j["BL"];
        for(auto it = data.begin(); it != data.end(); it++)
        {
               auto price = it->at("Price").get<double>(); // here raised type_error.302 
        }
```

i am not sure what is the issue, is the code has something wrong? and how to access array using our nlohmann API? thank you very much!

**Comment by nlohmann** [comment]
Can you print `data` to stdout please? I would assume `"BL"` does not exist, `j["BL"]` inserts a `null` value at that key which then explains the result.
